openproblems/tasks/dimensionality_reduction/methods/phate.py

```diff
@@ -16,34 +17,52 @@
 )
 
 
-def _phate(adata, test: bool = False, n_pca: Optional[int] = None, gamma: float = 1):
+def _phate(
+    adata, test: bool = False, genes=None, n_pca: Optional[int] = None, gamma: float = 1
+):
     from phate import PHATE
 
     if test:
         n_pca = n_pca or 10
     else:  # pragma: no cover
         n_pca = n_pca or 100
 
+    if genes is not None:
+        X = adata[:, genes].copy().X
+    else:
+        X = adata.X
+
     phate_op = PHATE(n_pca=n_pca, verbose=False, n_jobs=-1, gamma=gamma)
-    adata.obsm["X_emb"] = phate_op.fit_transform(adata.X)
+    adata.obsm["X_emb"] = phate_op.fit_transform(X)
     adata.uns["method_code_version"] = check_version("phate")
     return adata
 
 
 @_phate_method(method_name="PHATE (default)")
 def phate_default(adata, test: bool = False, n_pca: Optional[int] = None):
     adata = sqrt_cpm(adata)
-    return _phate(adata, test=test, n_pca=n_pca)
+    adata = _phate(adata, test=test, n_pca=n_pca)
+    # revert to expected adata.X
+    adata = log_cpm(adata)
```

Comment:
Does `X` get reset to counts somewhere or are we normalising twice by doing this?

Reply:
X is reset to counts in the dataset decorator as far as I'm aware.

Reply:
That's correct

Reply:
Ah, ok. I thought it must be happening somewhere but I couldn't find it.